src/stores/ontRoot.js

```diff
@@ -0,0 +1,101 @@
+import { defineStore } from 'pinia'
+import InstrumentFlowcellLayout from '@/config/InstrumentFlowcellLayout'
+import { handleResponse } from '@/api/ResponseHelper'
+import { groupIncludedByResource, dataToObjectById } from '@/api/JsonApi'
+import useRootStore from '@/stores'
+
+/**
+ * Generates an object describing the shared ONT resources for use in the vuex
+ * store
+ */
+
+// Helper function for reformatting data
+const formatById = (obj, data, includeRelationships = false) => {
+  return {
+    ...obj,
+    ...dataToObjectById({ data, includeRelationships }),
+  }
+}
+const useOntRootStore = defineStore('ontRoot', {
+  state: () => ({
+    // Resources returned by the server, each key represents a resource type.
+    // resource types are indexed by their id.
+    resources: {
+      // The main source of run information. runs are indexed by id.
+      runs: {},
+      // The main source of instrument information. instrument are indexed by id.
+      instruments: {},
+    },
+  }),
+  getters: {
+    runs: (state) => {
+      try {
```

Comment:
Same here. Not entirely sure what is happening so worth adding some documentation and again thinking about using types so something is always returned.